engine/gamesys/src/gamesys/scripts/script_sound.cpp

```diff
@@ -274,19 +268,17 @@ namespace dmGameSystem
      */
     static int Sound_GetGroups(lua_State* L)
     {
-        int top = lua_gettop(L);
+        DM_LUA_STACK_CHECK(L, 1);
+
+        dmhash_t groups[dmSound::MAX_GROUPS];
+        uint32_t count = dmSound::MAX_GROUPS;
+        dmSound::GetGroupHashes(&count, groups);
```

Comment:
Didn't make sense to have two functions with a lock each.
Replaced with one function, with a total of one lock.